src/EventStore.Client/EventStoreClientSettings.ConnectionString.cs

```diff
@@ -204,21 +220,45 @@ private static class ConnectionStringParser {
 					settings.ConnectivitySettings.TlsVerifyCert = (bool)tlsVerifyCert;
 				}
 
+				if (typedOptions.TryGetValue(TlsCaFile, out var tlsCaFile)) {
+					var tlsCaFilePath = Path.GetFullPath((string)tlsCaFile);
+					if (!string.IsNullOrEmpty(tlsCaFilePath) && !File.Exists(tlsCaFilePath)) {
+						throw new FileNotFoundException($"Failed to load certificate. File was not found.");
+					}
+
+					try {
+						using var x509Certificate2 = new X509Certificate2(tlsCaFilePath);
+						settings.ConnectivitySettings.TlsCaFile = tlsCaFilePath;
+					} catch (CryptographicException) {
+						throw new Exception("Failed to load certificate. Invalid file format.");
```

Comment:
It may be better to create a custom exception type, `InvalidClientCertificateException` and throw that, like so:
```csharp
throw new InvalidClientCertificateException("Invalid file format");
```

The exception we create can automatically prepend the string "Failed to load client certificate", too.

See the existing custom exceptions under `EventStore.Client.Exceptions.ConnectionString` for reference.

Reply:
Done 👍